x2gbfs/gbfs/gbfs_writer.py

```diff
@@ -75,4 +89,142 @@ def write_gbfs_feed(
         if alerts:
             feeds.append('system_alerts')
             self.write_gbfs_file(destFolder + '/system_alerts.json', {'alerts': alerts}, timestamp, ttl)
-        self.write_gbfs_file(destFolder + '/gbfs.json', self.gbfs_data(feed_language, base_url, feeds), timestamp, ttl)
+
+        if 'languages' in system_information:
+            feed_language = system_information['languages'][0]
+        elif 'language' in system_information:
+            feed_language = system_information['language']
+        else:
+            raise Exception('Config neither provides language nor languages to deduce feed language')
+
+        self.write_gbfs_file(destFolder + '/gbfs.json', self.gbfs_data(base_url, feeds, feed_language), timestamp, ttl)
+
+
+class GbfsV3Writer(GbfsV2Writer):
+
+    VEHICLE_STATUS_FEED_NAME = 'vehicle_status'
+    VEHICLE_STATUS_KEY = 'vehicles'
+
+    def write_gbfs_file(self, filename: str, data, timestamp: int, ttl: int = 60) -> None:
+        with open(filename, 'w') as dest:
+            content = {
+                'data': data,
+                'last_updated': datetime.utcfromtimestamp(timestamp).strftime('%Y-%m-%dT%H:%M:%SZ'),
+                'ttl': ttl,
+                'version': '3.0',
+            }
+            json.dump(content, dest, indent=2)
+
+    def gbfs_data(self, base_url: str, feeds: List[str], feed_language: str) -> Dict:
+        return {'feeds': [{'name': feed, 'url': f'{base_url}/{feed}.json'} for feed in feeds]}
+
+    def _convert_system_information_to_v3(self, system_information):
+        # This function converts a (possibly) v2 system_information into
+        # a v3 compliant system_information.
+        # If it's already a valid v3 config, it will be unchanged.
+
+        new_system_information = copy.deepcopy(system_information)
+        system_id = system_information['system_id']
+
+        if 'language' in new_system_information:
+            feed_language = new_system_information.pop('language')
+            new_system_information['languages'] = [feed_language]
+
+        for key in [
+            'name',
+            'attribution_organization_name',
+            'terms_url',
+            'privacy_url',
+            'operator',
+        ]:
+            if key in new_system_information and isinstance(new_system_information[key], str):
+                new_system_information[key] = [{'language': feed_language, 'text': new_system_information[key]}]
+
+        if 'phone_number' in new_system_information:
+            configured_phone_number = new_system_information['phone_number']
+            if not PHONE_NUMBER_GBFS_V3_PATTERN.match(configured_phone_number):
+                # Discard probably existing (and usually provided as alternative to a country code) "(0)"
+                phone_number_without_optional_prefix = configured_phone_number.replace('(0)', '')
```

Comment:
This implementation may work for those D/A/CH feeds that we know of. But it is dangerous to a) replace *all* `(0)` and b) assume that we can always safely remove even the first one.

> ### To make a number dialable, you only need to change the prefix
> In Argentina, to dial a mobile number domestically, the digits "15" need to be inserted after the area code but before the local number, and the "9" after the country code (54) needs to be removed. This transforms +54 9 2982 123456 into 02982 15 123456.

https://github.com/google/libphonenumber/blob/209b9d524bc375f57d100595b00e9b8dcd45defa/FALSEHOODS.md